### PR TITLE
Update to Python 3.11

### LINKED ISSUE #33: Handle alternative offset formats
Currently, you can specify an offset when setting a reminder, e.g.:

```
/remind 2023-02-14T20:00:00+01:00 Cry in bed because you're alone
```

However, the other common methods of specifying an offset (`+01` for just hours, or `+0100` for hours+minutes without a separator) do not work, because Python's "ISO" datetime parser is not ISO compliant.

Perhaps this should be detected and fixed manually within the parsing code in Lonabot? (Like we currently do for converting "Z" and "z" to "+00:00" because Python™.)

I'm prepared to work on this fix but not sure when I will have the time.

### PR DESCRIPTION
**Pull request by Sharparam**
This fixes #33.

### CODE REVIEW

File: lonabot/bot.py [hunk 1]
```diff
@@ -176,8 +176,8 @@ async def init(self):
             schedreminder.SchedReminder(r.id, r.due)
             for r in self.db.iter_reminders()
         )
-        self._check_sched_task = self._loop.create_task(self._check_sched())
-        self._check_bday_task = self._loop.create_task(self._check_bday())
```
Comment (9ary) [inline lonabot/bot.py:180]: Wait why was this still there?

Comment (9ary) [reply lonabot/bot.py:180]: Yeah #32 does the exact same thing???

Comment (Lonami) [reply lonabot/bot.py:180]: Mysteries from ~~the universe~~ git. I have no idea.

Comment (9ary) [reply lonabot/bot.py:180]: Nah @Sharparam's branch was a few commits behind master.

**Pull request merged**